File: Counting/3121. Count the Number of Special Characters II.cpp
class Solution {
public:
    int numberOfSpecialChars(string s) 
    {
        map<char,int>mp;
        int ans = 0, n = s.size();
        for(int i=0;i<n;i++) 
        {
            char ch = s[i];
            if(islower(ch)) mp[ch] = i+1;
            else if(mp[ch] == 0) mp[ch] = i+1;
        }
        
        for(char i='a';i<='z';i++)
        {
            int x = mp[i], y = mp[i - 'a' + 'A'];
            if(x and y and x < y) ans++;
        }
        return ans;
    }
};

File: Counting/3121. Count the Number of Special Characters II.py
class Solution:
    def numberOfSpecialChars(self, word: str) -> int:
        n = len(word)
        store = defaultdict(int)
        for i in range(n):
            if word[i].islower():
                store[word[i]] = i
            elif word[i] not in store:
                store[word[i]] = i

        ans = 0
        for ch in ascii_lowercase:
            CH = ch.upper()
            if ch in store and CH in store:
                ans += store[ch] < store[CH]
        return ans